### PR TITLE
Remove OLM integration as we aren't using it.

### PR DESCRIPTION
**Pull request by awels**
Removed OLM code since it is not being used in CDI deployment.

Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Removes OLM catalog and CSV generation as we aren't using it and it is causing problems.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by awels** [comment]
/hold
Need to update hack/README.md and create a new deploy readme in docs.

**Comment by awels** [comment]
If we keep the CSV generation stuff, then we need to keep the parameters passed into the generator. And we have to keep them at some value. And I prefer to not keep cruft around if we aren't using it. We have all the code in git, so if we need to retrieve it, its easy to find.

**Comment by awels** [comment]
@mhenriks Checked with Ryan and David and yes they do use the csv generator, so I put that code back.

File: hack/build/build-copy-artifacts.sh [hunk 1]
```diff
@@ -31,12 +31,6 @@ for tgt in ${targets}; do
     if [[ "${bin_name}" == "${CLONER}" ]]; then
         mkdir -p "${CMD_OUT_DIR}/${bin_name}/"
         cp -f "${CDI_DIR}/cmd/${bin_name}/cloner_startup.sh" "${dest_dir}"
-    elif [[ "${bin_name}" == "${OPERATOR}" ]]; then
-        # This container also needs the csv-generator file, so it needs to be
-        # copied into the operator's directory for docker build to source
-        # csv-generator correctly
-        mkdir -p "${CMD_OUT_DIR}/${bin_name}/"
-        cp ${OUT_DIR}/tools/${CSV_GENERATOR}/${CSV_GENERATOR} ${OUT_DIR}/cmd/${OPERATOR}/${CSV_GENERATOR}
```
Comment (mhenriks) [inline hack/build/build-copy-artifacts.sh]: doesn't this have to stay?

File: hack/build/config.sh [hunk 2]
```diff
@@ -19,15 +19,13 @@ APISERVER="cdi-apiserver"
 UPLOADPROXY="cdi-uploadproxy"
 UPLOADSERVER="cdi-uploadserver"
 OPERATOR="cdi-operator"
-CDI_OLM_CATALOG="cdi-olm-catalog"
 FUNC_TEST_INIT="cdi-func-test-file-host-init"
 FUNC_TEST_HTTP="cdi-func-test-file-host-http"
 FUNC_TEST_REGISTRY="cdi-func-test-registry"
 FUNC_TEST_REGISTRY_POPULATE="cdi-func-test-registry-populate"
 FUNC_TEST_REGISTRY_INIT="cdi-func-test-registry-init"
-CSV_GENERATOR="csv-generator"
```
Comment (mhenriks) [inline hack/build/config.sh]: should this stay?

File: hack/build/docker/cdi-operator/Dockerfile [hunk 3]
```diff
@@ -17,6 +17,5 @@ WORKDIR /home/cdi-operator
 USER 1001
 
 COPY ./cdi-operator /usr/bin/cdi-operator
-COPY ./csv-generator /usr/bin/csv-generator
```
Comment (mhenriks) [inline hack/build/docker/cdi-operator/Dockerfile]: Should this stay?

**Comment by mhenriks** [comment]
Should we add a functional test that executes the csv-generator?

File: hack/build/build-copy-artifacts.sh (hunk 1 above)
Comment (awels) [reply hack/build/build-copy-artifacts.sh]: Well only the downstream version is used, and the downstream build still has the information needed to compile and copy the binary into the downstream container. They shouldn't be using the upstream csv generator to create the HCO CSV.

File: hack/build/config.sh (hunk 2 above)
Comment (awels) [reply hack/build/config.sh]: Same answer, if we don't care about upstream CSV generator, then we don't need to compile and install it in the container.

File: hack/build/docker/cdi-operator/Dockerfile (hunk 3 above)
Comment (awels) [reply hack/build/docker/cdi-operator/Dockerfile]: Same answer, if we don't care about upstream CSV generator, then we don't need to compile and install it in the container.

Comment (awels) [reply hack/build/docker/cdi-operator/Dockerfile]: Only reason I can see it remaining here, is to stop the Dockerfile check downstream from triggering that something changed.

**Pull request merged**